### PR TITLE
Add program_type and program_names to State model

### PR DESCRIPTION
**Pull request by tadhg-ohiggins**
Program type and names
Belong to states not users.
Add them to the states.

- Add `program_type` and `program_names` to `State` model.
- Add list of `program_type` options.
- Add `State`, `ACS`, `FMAP`, and `AppUser` to models visible in Django admin.
- Use `state-fixture-data.csv` for populating the state fixtures with the new program_type and program_names fields.
- Remove American Samoa, Guam, the Northern Mariana Islands, Puerto Rico, and the Virgin Islands from the fixture data, because none of them report to CARTS.

This is related to #206 because separating what's associated with states versus users is necessary before developing the models/endpoints, and users are going to be associated with states and pull program type and program names from the states, so we need to populate that info.

### CODE REVIEW

**Comment by tadhg-ohiggins** [comment]
FWIW I did test this against a running database with prior data in it rather than a completely new database, and the migrations and fixtures worked fine, but that's not a guarantee that that'll work in staging…

**Comment by tadhg-ohiggins** [comment]
@kaitlin @erinzstrenio @StephanieCoppel I removed American Samoa, Guam, the Northern Mariana Islands, Puerto Rico, and the Virgin Islands from the FMAP and state data, let me know if we need to add them back for some reason.

**Comment by StephanieCoppel** [comment]
> @kaitlin @erinzstrenio @StephanieCoppel I removed American Samoa, Guam, the Northern Mariana Islands, Puerto Rico, and the Virgin Islands from the FMAP and state data, let me know if we need to add them back for some reason.

@tadhg-ohiggins we do not need to add those back, thanks for removing those. thanks!

File: frontend/api_postgres/carts/carts_api/models.py [hunk 1]
```diff
@@ -40,30 +43,64 @@ class State(models.Model):
...
+    number_uninsured = models.IntegerField(
+        help_text="the number of uninsured children under 200% FPL"
+    )
+    number_uninsured_moe = models.IntegerField(
+        help_text="the margin of error for the number of uninsured children"
+    )
+    # help_text has exactly five spaces between “uninsured” and “children” in
+    # order to avoid triggering a database migration for such a minor issue.
+    percent_uninsured = models.DecimalField(
+        help_text="percentage of uninsured     children",
```
Comment (mgwalker) [inline frontend/api_postgres/carts/carts_api/models.py:94]: The result of reformatting the code, I'm guessing? Does the backslash preserve whitespace?

**Review by mgwalker — APPROVED** [review]
Looks fine to me, but having @timoballard's eyes on `generate_fixtures` wouldn't be the worst idea.

**Review by timoballard — APPROVED** [review]

File: frontend/api_postgres/carts/carts_api/model_utils.py [hunk 2]
```diff
@@ -61,6 +67,6 @@
 )
 
 USER_ROLES = (
-    ("state_user", "State User"),
     ("co_user", "Central Office User"),
+    ("state_user", "State User"),
```
Comment (timoballard) [inline frontend/api_postgres/carts/carts_api/model_utils.py:71]: this reordering results in a new migration being generated - I think we should either back this change out or commit the migration

**Review by timoballard — CHANGES_REQUESTED** [review]

File: frontend/api_postgres/carts/carts_api/model_utils.py (hunk 2 above)
Comment (tadhg-ohiggins) [reply frontend/api_postgres/carts/carts_api/model_utils.py:71]: I'll back it out.

**Comment by tadhg-ohiggins** [comment]
@timoballard Could you verify that this version doesn't trigger a migration? If it's good to go after that, feel free to merge it in.

**Comment by jgillis1** [comment]
As a note, please be aware that states can change/add/remove programs year over year. So the 2020 program name and type could be different in 2022 (i.e. the 2020 program is discontinued and replaced with a new program). Does that need to be accounted for in this model or would it be handled in the database only?

**Comment by timoballard** [comment]
@tadhg-ohiggins - just checked, no migrations now 👍

**Comment by tadhg-ohiggins** [comment]
@jgillis1 Yes, I was thinking about this—do we need to track state historical program type, or would it be enough to have that info in their answers for previous years? So if MA is a combo state in 2020 but a separate_chip state in 2021, would it be enough for their CARTS report itself to have that data but their state info in the DB say they're separate_chip? I was assuming that having the info in the reports would be sufficient. @StephanieCoppel any thoughts on this?

**Comment by jgillis1** [comment]
We probably don't (though the current system does) but Program Type drives
behavior in the app in terms of what is shown and not shown in some
sections. You're saying that the form then would be the source of truth for
program type and name rather than what you are loading into the state
model? And, follow up question, if that is so, what is the purpose of
adding them to the state model? (Just curious there)


*Jenna Gillis*
Full Stack Developer, ACME Dev Team Lead
CollabraLink Technologies, Inc.
jgillis@collabralink.com
*C:* 443.807-7580

**Comment by jgillis1** [comment]
@tadhg-ohiggins We probably don't need it historically (though the current system does) but Program Type drives behavior in the app in terms of what is shown and not shown in some sections. You're saying that the form then would be the source of truth for program type and name rather than what you are loading into the state model? And, follow up question, if that is so, what is the purpose of adding them to the state model? (Just curious there)

**Comment by tadhg-ohiggins** [comment]
The state model would be the source of truth for active forms. So if MA is combo in 2020 that would be shown in the state model. But if MA had been separate_chip in any prior year, that would only be captured in the MA report for that year.

**Review by timoballard — APPROVED** [review]

**Comment by jgillis1** [comment]
@tadhg-ohiggins Could you add comments to the model RE: the above so the set up is easy to find and reference in the future?

**Pull request merged**